keyboard/src/lib/y2country/widgets.rb

```diff
@@ -0,0 +1,74 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LINUX GmbH, Nuernberg, Germany.
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may find
+# current contact information at www.novell.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "cwm/widget"
+
+Yast.import "Keyboard"
+
+module Y2Country
+  class KeyboardSelectionWidget < CWM::SelectionBox
+    def initialize
+      textdomain "country"
+    end
+
+    def label
+      # widget label
+      _("&Keyboard Layout")
+    end
+
+    def init
+      if Yast::Keyboard.user_decision
+        self.value = Yast::Keyboard.current_kbd
+      else
+        self.value = "english-us"
```

Comment:
This seems out of place. The rest of the code does not care about how a keyboard ID looks like or what the default is. Can we inject it from the outside?

BTW the IDs are yast specific, defined in [/usr/share/YaST2/data/keyboard_raw.ycp](https://github.com/yast/yast-country/blob/a0bf4289fb27795da0840a4cbff724e82a19f8ac/keyboard/src/data/keyboard_raw.ycp)